synth_node_bin/src/action/advanced_sn_for_s001.rs

```diff
@@ -0,0 +1,181 @@
+use std::{net::SocketAddr, str::FromStr};
+
+use anyhow::Result;
+use tokio::time::{interval, sleep, Duration};
+use ziggurat_zcash::{
+    protocol::{
+        message::Message,
+        payload::{addr::NetworkAddr, Addr, Nonce},
+    },
+    tools::{
+        message_filter::{Filter, MessageFilter},
+        synthetic_node::SyntheticNode,
+    },
+};
+
+use super::{ActionCfg, SynthNodeAction};
+
+// Configurable status printout interval.
+const DBG_INFO_LOG_INTERVAL_SEC: Duration = Duration::from_secs(10);
+const BROADCAST_INTERVAL_SEC: Duration = Duration::from_secs(120);
+
+pub(super) struct Action;
+
+pub(super) fn action() -> Box<dyn SynthNodeAction> {
+    Box::new(Action {})
+}
+
+#[async_trait::async_trait]
+impl SynthNodeAction for Action {
+    fn info(&self) -> &str {
+        "an advanced SN node created for RT-S001 which can do the following
+           - listens for inbound connections and prints the connection status periodically
+           - periodically requests and sends Ping/GetAddr/Addr messages to all peers"
+    }
+
+    fn config(&self) -> ActionCfg {
+        ActionCfg {
+            msg_filter: MessageFilter::with_all_auto_reply().with_getaddr_filter(Filter::Disabled),
+        }
+    }
+
+    #[allow(unused_variables)]
+    async fn run(&self, synth_node: &mut SyntheticNode, addr: SocketAddr) -> Result<()> {
+        // Sleep for 3 seconds before taking any actions.
```

Comment:
What's the reason for slipping 3 secs? 
Now, this comment just duplicate the code and state the obvious, telling just what the next line is doing.
Maybe it would make the comment meaningful when it could include some justification about the chosen delay here.

Reply:
Done